### PR TITLE
Sync develop changes April 4 -April 22 to hdf5_1_14

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

**Comment by lrknox** [comment]
I'm finding /develop/ in some paths to documentation.  Many of them should replaced with /v1_14/.  I'll be adding a new commit with those changes later this afternoon.

File: doxygen/dox/ExamplesAPI.dox [hunk 1]
```diff
@@ -387,347 +387,345 @@ FORTRAN
...
-<td><a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/C/H5T/tfiles/18/h5ex_t_float.ddl">h5ex_t_float.ddl</a></td>
+<td><a href="https://\SRCURL/HDF5Examples/C/H5T/tfiles/16/h5ex_t_float.tst">h5ex_t_float.tst</a></td>
+<td><a href="https://\SRCURL/HDF5Examples/C/H5T/tfiles/18/h5ex_t_float.ddl">h5ex_t_float.ddl</a></td>
 </tr>
 <tr>
 <td>Read / Write Integer Datatype (Attribute)</td>
 <td>
-<a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/C/H5T/h5ex_t_intatt.c">C</a>
-<a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/FORTRAN/H5T/h5ex_t_intatt_F03.F90">FORTRAN</a>
-<a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/JAVA/H5T/H5Ex_T_IntegerAttribute.java">Java</a>
```
Comment (jhendersonHDF) [inline doxygen/dox/ExamplesAPI.dox:561]: Looks like this link to the Java example was dropped here rather than converted to the `\SRCURL` approach

Comment (lrknox) [reply doxygen/dox/ExamplesAPI.dox:561]: added link in new commit, along with changes from /develop/ to /v1_14/ or /hdf5_1_14/ in some .md files.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ Please make sure that you check the items applicable to your pull request:
   * [ ] If changes were done to Autotools build, were they added to CMake and vice versa?
   * [ ] Is the pull request applicable to any other branches? If yes, which ones? Please document it in the GitHub issue.
   * [ ] Is the new code sufficiently documented for future maintenance?
-  * [ ] Does the new feature require a change to an existing API? See "API Compatibility Macros" document (https://docs.hdfgroup.org/hdf5/v1_14/api-compat-macros.html)
+  * [ ] Does the new feature require a change to an existing API? See "API Compatibility Macros" document (./api-compat-macros.html)
```
Comment (jhendersonHDF) [inline CONTRIBUTING.md]: Does this one need to be changed back to a full link instead of a "./"?

File: release_docs/RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    - Each resolved GH milestone issue traces to one or more revisions to the HDF5 source.
+    - Each resolved GH milestone issue traces to one or more pull requests.
+2. For each previously authored KNOWN ISSUE in the [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/release_docs/RELEASE.txt), if the issue has been resolved or can no longer be confirmed, remove the issue from the [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/release_docs/RELEASE.txt).
+    - Document any new known issues at the top of the list.
+3. Update the TESTED CONFIGURATION FEATURES SUMMARY in [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/release_docs/RELEASE.txt) to correspond to features and options that have been tested during the maintenance period by the automated daily regression tests. 
+    - **See: Testing/Testing Systems(this is a page in confluence)**
+4. Update current compiler information for each platform in the PLATFORMS TESTED section of [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdff5_1_14/release_docs/RELEASE.txt).
```
Comment (jhendersonHDF) [inline release_docs/RELEASE_PROCESS.md]: Markdown checker failing here. `hdff5_1_14` -> `hdf5_1_14`

File: release_docs/RELEASE_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+5. Review the [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/release_docs/RELEASE.txt) for formatting and language to verify that it corresponds to guidelines found in **[Writing Notes in a RELEASE.txt(this is missing)]()** File.
+6. Review and update, if needed, the [README](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/README.md) and [COPYING](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/COPYING) files.
+7. Review and update all INSTALL_* files in [release_docs](https://github.com/HDFGroup/hdf5/tree/hdf5-1_14/release_docs), if needed.
```
Comment (jhendersonHDF) [inline release_docs/RELEASE_PROCESS.md]: Markdown checker failing here. I think `hdf5-1_14` -> `hdf5_1_14`?

File: CONTRIBUTING.md (hunk 2 above)
Comment (lrknox) [reply CONTRIBUTING.md]: Yes, fixed

File: release_docs/RELEASE_PROCESS.md (hunk 3 above)
Comment (lrknox) [reply release_docs/RELEASE_PROCESS.md]: fixed

File: release_docs/RELEASE_PROCESS.md (hunk 4 above)
Comment (lrknox) [reply release_docs/RELEASE_PROCESS.md]: yes, corrected

**Review by byrnHDF — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**